timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/LockEventLogImpl.java

```diff
@@ -64,7 +78,22 @@ public synchronized <T> ValueAndLockWatchStateUpdate<T> runTask(
     @Override
     public synchronized void logLock(
             Set<LockDescriptor> locksTakenOut, LockToken lockToken, Optional<LockRequestMetadata> metadata) {
-        slidingWindow.add(LockEvent.builder(locksTakenOut, lockToken, metadata));
+        Optional<LockWatchEvent> replacedEvent =
```

Comment:
Open question: Do we think we should have the metric recorded here, or in `ArrayLockEventSlidingWindow`? What's sort of frustrating by the previous implementation, is that we have the storage of lock events, and then the processing. Feels a bit weird to me that we track the storage of lock events at the processing layer, although the same times it sort of makes sense, as we avoid wiring in multiple places.

That said, maybe we can use the delegate pattern to help us?

Reply:
I think delegating the metric tracking can be worthwhile if we do track a lot but for 2-3 metrics I don't think it's worth it yet